### PR TITLE
Being drunk or having a broken arm hurts accuracy with ranged weapons. Also cleans up a fair bit of hand handling to be more readable

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

1. Having a broken arm will affect your accuracy when firing a weapon with that arm, even potentially causing damage to you if it's a weapon with recoil. This effect is not applied under the effects of a painkiller. 

2. The sister effect of this (punching someone with a broken arm) now also has an interact with painkillers (it can no longer block your attack).

3. Being drunk heavily affects ranged weapon accuracy, unless you're the bartender

4. A lot of hand handling cleanup, using new macros to make it a lot more readable at a glance 

## Why It's Good For The Game

We have this system for modifying firearm accuracy but we don't really use it commonly, and I feel like it slots in well with a lot of places

For broken arms, it adds some more depth to the wound system, in the same way that trying to punch someone with a broken arm causes pain. (I actually want to expand this to melee weapon accuracy and attacking with melee weapons in general, but that's for a later time)

For drunkenness, it just adds to the drunk shenanigans. It also slightly reduces the effectiveness of drinks as combat healing chemicals, such as quadsec - makes it a bit more of a trade off.

## Changelog

:cl: Melbert
balance: Having a broken arm affects your accuracy with ranged weapons fired with that arm. Utilizing a painkiller will nullify this effect, however. 
balance: Painkillers will prevent your punches from being cancelled due to having a broken arm. You'll still take damage, though.
balance: Being drunk now affects your accuracy with ranged weapon. The bartender is immune to this effect via their skillchip.
code: A lot of code involving left and right hand handling has been cleaned up, easier to read. Report any oddities, like left and rights being flipped
/:cl:

### CODE REVIEW

**Comment by Xkallubar** [comment]
Make it not hurt accuracy with grenade launchers for the funny tf2 reference.

**Comment by MrMelbert** [comment]
> Make it not hurt accuracy with grenade launchers for the funny tf2 reference.

image

File: code/__DEFINES/inventory.dm [hunk 1]
```diff
@@ -135,6 +135,10 @@ DEFINE_BITFIELD(no_equip_flags, list(
 //defines for the index of hands
 #define LEFT_HANDS 1
 #define RIGHT_HANDS 2
+/// Checks if the value is "left" - same as ISEVEN, but used primarily for hand or foot index contexts
+#define IS_RIGHT(value) (value % 2 == 0)
+/// Checks if the value is "right" - same as ISODD, but used primarily for hand or foot index contexts
+#define IS_LEFT(value) (value % 2 != 0)
```
Comment (Kapu1178) [inline code/__DEFINES/inventory.dm:141]: muh namespace, looking at you `INFINITE` 
```suggestion
#define IS_RIGHT_INDEX(value) (value % 2 == 0)
/// Checks if the value is "right" - same as ISODD, but used primarily for hand or foot index contexts
#define IS_LEFT_INDEX(value) (value % 2 != 0)
```

**Comment by Exester509** [comment]
Why? It's perfectly safe to discharge guns while drunk, if you can drive a car (and you can) you can shoot a gun.

**Comment by MrMelbert** [comment]
> Why? It's perfectly safe to discharge guns while drunk, if you can drive a car (and you can) you can shoot a gun.

Just wait until I add bring drunk causing you to crash shuttles 2 tiles into the station

File: code/datums/status_effects/debuffs/drunk.dm [hunk 2]
```diff
@@ -123,6 +124,19 @@
...
+	UnregisterSignal(owner, COMSIG_MOB_FIRED_GUN)
+
+/datum/status_effect/inebriated/drunk/proc/drunk_gun_fired(datum/source, obj/item/gun/gun, atom/firing_at, params, zone, bonus_spread_values)
+	SIGNAL_HANDLER
+
+	// excusing the bartender, because shotgun
+	if(HAS_TRAIT(owner, TRAIT_DRUNKEN_BRAWLER))
+		return
+	// what makes me a good demoman?
+	if(istype(gun, /obj/item/gun/grenadelauncher) || istype(gun, /obj/item/gun/ballistic/revolver/grenadelauncher))
```
Comment (Ghommie) [inline code/datums/status_effects/debuffs/drunk.dm:136]: This vould be an item trait.

File: code/modules/antagonists/heretic/knowledge/side_void_blade.dm [hunk 3]
```diff
@@ -107,22 +107,6 @@
 	demolition_mod = 1.5
 	sharpness = SHARP_EDGED
 
-/obj/item/mutant_hand/shattered_risen/Initialize(mapload)
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/knowledge/side_void_blade.dm:110]: Is this removal from here to line 125 intentional?

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/heretic/knowledge/side_void_blade.dm (hunk 3 above)
Comment (MrMelbert) [reply code/modules/antagonists/heretic/knowledge/side_void_blade.dm:110]: Yes, all of it is copy pasted on the base type

**Comment by Ghommie** [comment]
It doesn't compile.

**Pull request merged**